src/main/java/com/gru/ifsp/AgendamentoBanca/services/AgendamentoBancaServiceImpl.java

```diff
@@ -41,36 +39,67 @@ public AgendamentoBancaServiceImpl(AgendamentoRepository agendamentoRepository,
-
-        AgendamentoBanca agendamentoBanca = AgendamentoBancaUtils.convertFormToAgendamentoBanca(form, listaAlunos, listaProfessores);
+       AgendamentoBanca agendamentoBanca = AgendamentoBancaUtils.convertFormToAgendamentoBanca(form, listaAlunos, listaProfessores);
 
         agendamentoRepository.save(agendamentoBanca);
         //Add users on banca through the entity responsible for assert relationship beetween banca and users
         addUsuariosOnBanca(agendamentoBanca, listaAlunos, false);
         addUsuariosOnBanca(agendamentoBanca,listaProfessores, true);
-
+        //Set who is admin of banca
+        setAdminsOfBanca(agendamentoBanca,form.getAdminsBanca());
         return agendamentoBanca;
     }
 
-    private void checkyIfCanCreateAgendamentoOnThisTime(String dataAgendamento) throws RuntimeException {
+    private void setAdminsOfBanca(AgendamentoBanca banca, Long[] adminsBanca) {
+        List<UsuarioParticipantesPorBanca> listaProfessores = usuariosParticipantesPorBancaRepository.findAllByBancaIsAndIsTeacher(banca, true);
+        if(listaProfessores.size() > 0){
+            listaProfessores.forEach(professor -> Arrays.stream(adminsBanca).forEach(id -> {
+                if (Objects.equals(id, professor.getUsuario().getId())) {
+                    professor.setIsAdmin(true);
+                }
+            }));
+            usuariosParticipantesPorBancaRepository.saveAll(listaProfessores);
+        } else{
+            throw new RuntimeException("Não há usuários permitidos para admistrar bancas");
+        }
+    }
+
+    public void updateUserForAdmin(Long idBanca, Long idUsuario, boolean permission){
+        var banca = getById(idBanca);
+        var usuario = userRepository.findById(idUsuario).orElseThrow(UsuarioNaoEncontradoNaBanca::new);
+        var usarioOnBanca = usuariosParticipantesPorBancaRepository
+                .findByBancaAndUsuario(banca, usuario).get();
+
+        usarioOnBanca.setIsAdmin(permission);
+        usuariosParticipantesPorBancaRepository.save(usarioOnBanca);
+    }
+
+    private LocalDateTime StringDataToLocalDateTime(String date){
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
-        var data = LocalDateTime.parse(dataAgendamento, formatter);
+        return LocalDateTime.parse(date, formatter);
+    }
 
+    private void checkyIfCanCreateAgendamentoOnThisTime(String dataAgendamento) throws RuntimeException {
+        var data = DateStringToLocalDateTime(dataAgendamento);
 
         var horarios = agendamentoRepository
                 .findByDataAgendamentoBetween(data.minusMinutes(50),
                         data.plusMinutes(50));
 
         if(horarios.size() > 0){
-            throw new RuntimeException("Banca não pode ser cadastrada neste horário");
+           for(var horario : horarios){
+               if(!(data == horario.getDataAgendamento()) && horarios.size() > 1){
```

Comment:
O melhor jeito seria vendo se a banca que ele achou não é a mesma banca do formulário. Nesse caso, se tiver somente 1 banca com esse horário e essa banca for a mesma que a do formulário, essa ele poderia desconsiderar.